### PR TITLE
Closes #98 -  minor word changes in README.md

### LINKED ISSUE #98: General Issue: Add comments to raw-data/create_adams_data.R?
### Background Information

This is good code, but might benefit from a small number of comments.

This script locates and runs **ADaM templates** in the various **admirals** packages.  
A few comments would aid in future maintenance.

Examples:
- system.file(package = 'pkg')  returns? [depends on whether load_all() has run]
- generally clarify which **files** and **paths** [often it is where the package is **installed**]

### Definition of Done

no more than a few comments

### PR DESCRIPTION
**Pull request by jimrothstein**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build pharmaverseadam site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new datasets occur on the "[Reference](https://pharmaverse.github.io/pharmaverseadam/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jimrothstein** [comment]
@manciniedoardo
Added only a few comments, just to start.
README.md is very good; you are correct.

**Comment by jimrothstein** [comment]
Not right approach.   I have another idea ....

**Comment by jimrothstein** [comment]
main
Better approach?

**Comment by jimrothstein** [comment]
cmd is of the form ...
run_template png

**Comment by jimrothstein** [comment]
@manciniedoardo 

Not ready for true review.   

- **Goal: document script that creates ADaM files.**
Idea was to `add comments`.  But for my own understanding, I find annotating lines -- see above -- to be clearest.  Also, several directories are involved.

- **So what I'd like to do is explain the `big picture` first**  in 2-3 sentences and list the `directories` to be used.
Then code comments  might be easier to follow.

If  OK, I'll go ahead and do this.

**Comment by jimrothstein** [comment]
@manciniedoardo 
Appreciate if you take a look, but I DO NEED to go through comments again.  
So asking what you think so far. 
(Some comments are not mine, some comments need to cleaned,  need to redo the introduction).

File: README.md [hunk 1]
```diff
@@ -11,11 +11,11 @@ Test data (ADaM) for the pharmaverse family of packages
 
 # Purpose
 
-To provide a one-stop-shop for ADaM test data in the pharmaverse family of packages.
+To provide a one-stop-shop to create ADaM test data in the pharmaverse family of packages.
```
Comment (manciniedoardo) [inline README.md:14]: I don't think this is an improvement in the language. Reasoning: pharmaverseadam **already has** the test data, the expectation is that users just use the data from here.

File: README.md [hunk 2]
```diff
@@ -11,11 +11,11 @@ Test data (ADaM) for the pharmaverse family of packages
...
 
 # Package Contents
 
-The ADaM contents of this package is populated by an action that executes the `{admiral}`, `{admiralonco}`, `{admiralophtha}`, `{admiralvaccine}` and `{admiralpeds}` templates and saves the resulting datasets here. This action can be triggered manually by the package maintainers in two scenarios:
+The ADaM datasets of this package are populated by an action that executes the `{admiral}`, `{admiralonco}`, `{admiralophtha}`, `{admiralvaccine}` and `{admiralpeds}` templates and saves the resulting datasets here. This action can be triggered manually by the package maintainers in two scenarios:
```
Comment (manciniedoardo) [inline README.md:18]:
```suggestion
The ADaM datasets in this package are populated by an action that executes the `{admiral}`, `{admiralonco}`, `{admiralophtha}`, `{admiralvaccine}` and `{admiralpeds}` templates and saves the resulting datasets here. This action can be triggered manually by the package maintainers in two scenarios:
```

File: data-raw/create_adams_data.R [hunk 3]
```diff
@@ -109,7 +134,7 @@ run_template <- function(tp) {
       # write labels
       data <- write_labels(data, dataset_name, suffix)
 
-      # save it to pharmaverseadam data package
+      # save file to pharmaverseadam data dir (pharmaverseadam/data/adae.rda)
```
Comment (manciniedoardo) [inline data-raw/create_adams_data.R:137]:
```suggestion
      # save file to pharmaverseadam data dir (pharmaverseadam/data/<adamname>.rda)
```

File: data-raw/create_adams_data.R [hunk 4]
```diff
@@ -160,26 +186,29 @@ for (pkg in packages_list) {
       upgrade = "always", force = TRUE
     )
   }
-  # get templates scripts
+  # find templates scripts and rum them
```
Comment (manciniedoardo) [inline data-raw/create_adams_data.R:189]:
```suggestion
  # find templates scripts and run them
```

File: data-raw/create_adams_data.R [hunk 5]
```diff
@@ -160,26 +186,29 @@ for (pkg in packages_list) {
...
+  #
+  # find path to ADaM templates for the installed pkg (ex: ~/R/86_64...library/4.4/admiral/templates)
   templates_path <- file.path(system.file(package = pkg), "templates")
+  # templates are R files with a prefix `ad_` (1 or more files)
   templates <- list.files(templates_path)
   # copy paste pkg/data folder content to pharmaverseadam/data (some templates have dependency with their internal data,
   # for example admiral templates have all dependencies with admiral_adsl dataset)
+
+
```
Comment (manciniedoardo) [inline data-raw/create_adams_data.R:198]: i think  we can keep this because it's useful for logging purposes

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @jimrothstein! Left some comments

File: README.md (hunk 1 above)
Comment (jimrothstein) [reply README.md:14]: you are correct;   I restored original language.

File: README.md (hunk 2 above)
Comment (jimrothstein) [reply README.md:18]: agreed

File: data-raw/create_adams_data.R (hunk 3 above)
Comment (jimrothstein) [reply data-raw/create_adams_data.R:137]: I will adopt this your notation (  pharmaverseadam/data/<adamname>.rda)

**Comment by jimrothstein** [comment]
@manciniedoardo 
Made your changes ... thanks!

A few more changes/comments will  make the code clearer, I think.
- improve notation for files/directories.
- a few more clarifying comments.
- some prior comments are not helpful to me.

But first, `do you feel it is worth the effort`?

**Comment by manciniedoardo** [comment]
> @manciniedoardo Made your changes ... thanks!
> 
> A few more changes/comments will make the code clearer, I think.
> 
> * improve notation for files/directories.
> * a few more clarifying comments.
> * some prior comments are not helpful to me.
> 
> But first, `do you feel it is worth the effort`?

thanks @jimrothstein! absolutely, feel free to take it as far as you deem necessary.

**Comment by jimrothstein** [comment]
@manciniedoardo     **UPDATE**

An issue in `admiral` is to verify templates are working.  (https://github.com/pharmaverse/admiral/pull/2662)  
This PR `pharamaverseadam` uses same templates to generate `ADaMs`.

Working with @StefanThoma on admiral issue. 

Where the two scripts overlap, I hope  to make variables, functions `consistent`  .   

Not there yet.   When admiral script is more stable, will ask for your advice and suggestions.

**Comment by manciniedoardo** [comment]
@jimrothstein we just merged a small update to the documentation process that will have created a conflict, please merge main into this branch and accept the changes from main

**Comment by manciniedoardo** [comment]
@jimrothstein @StefanThoma maybe it would be good for us to meet at some point soon and discuss how this is going/next steps?

**Comment by manciniedoardo** [comment]
> can we please update this branch from main, and run a reverse dependency check before approving this thanks

sure, but at the same time I'm not sure of the status of this PR anyway as `create_adams_data.R` file has been edited other times and for parallel efforts anyway.
